1a)  
  1) values added: 20 is printed by line 9. This is printed since we can access the var result value throughout the entire function. We just update the variable accordingly and print it out.
  2) final result: 20 is printed by line 13. Again, this is printed since we can access the var result value throughout the entire function, we can have access to it even after the if statement block. Thus, we just update the variable accordingly and print it out.
  3) values added: 20 is printed by line 9. This is printed because the let result variable is declared within the if statement block and the print statement in line 9 is also within the same if statement block. So since they are in the same block, we can just update the variable accordingly and print it out. 
  4) The code will return an error because the let result variable is declared in the if statement block & line 13 is outside of the if statement block. And we know that when you declare a variable with the let keyword, that variable only has block scope so we won't be able to access result and print it.  
  5) The code will return an error because you aren't supposed to reassign the result value after it is set as 0 since it is of type const.   
  6) Again, the code will return an error because you aren't supposed to reassign the result value after it is set as 0 since it is of type const. Since we call sumValues(10,10, true), we will run the if statement which will attempt to change the value of result, but we can't do that since it is a const.  

1b) 
  1) In line 12, we will print 3. Since i is a var, we know that we can access it throughout the entire function. So since we are printing at line 12 (which is within the discountPrices function definition), we will not get an error. We get the value 3 for i by simply going through the for loop and updating it accordingly & eventually printing the value of i at line 12. 
  2) In line 13, we will print 150. Again, since discountedPrice is of type var, we know that we can access it throughout the entire function. So since we are printing the value of discountedPrice at line 13 (which is within the discountPrices function definition), we will not get an error. We get the value 150 by continually updating the value of discountedPrice as we go through the for loop. And we end up with 150 because that is the last updated value of discountedPrice after we exit the for loop.
  3) In line 14, we will also print 150. Since finalPrice is of type var, we know that we can access it throughout the entire function (it is also not defined within a certain block, it's defined at the top of the function). So since we are printing the value of finalPrice at line 14 (which is within the discountPrices function definition), we will not get an error. We get the value 150 by continually updating the value of discountedPrice as we go through the for loop. And we end up with 150 because that is the last updated value of finalPrice after we exit the for loop.
  4) This function will return a new array called discounted with the updated discountedPrice values. It will return [50, 100, 150]. We return the discounted array which again is of type var, so we can access it throughout the entire function - which is why we don't get an error. And we end up with these values in the array because in the discountPrices function, we have a variable called discountedPrice that we continually update as we go through the for loop & push that value into the discounted array for each iteration of the for loop so that the array holds each of the discountedPrices.
  5) This will cause an error because the variable i is of type let and since we declare it within the for loop block, we will not be able to access it at line 12 when we want to print the value of i. 
  6)  Again, this will cause an error because the variable discountedPrice is of type let and since we declare it within the for loop block, we will not be able to access it at line 13 when want to print the value of discountedPrice.
  7)  At line 14, we will print the value 150. It will not cause an error because even though the finalPrice variable is of type let, it is not declared within a specific block of code in the function. finalPrice is declared at the top of the discountPrices function and is not within a block, so when we want to print the value of it at line 14, we can do so without error. We get the value 300.5 by simply going through the for loop and updating the variable accordingly. We end up with 150 because that is the last updated value of finalPrice after we exit the for loop.
  8) This function will still return a new array called discounted with the updated discountedPrice values. It will return [50,100,150]. Even though this array is of type let, it is again not declared within a specific block of code in the function. The discounted array is declared at the top of the discountPrices function is not declared within a block, so when we reference it at the end of the function to return it, we won't get an error. And we get all the correct values within this array since we push the updated values of the discountedPrice into the discounted array as we go through the for loop.
  9) At line 11, the code will cause an error. Because again, the variable i is of type let and since we declare it within the for loop block, we will not be able to access it at line 11 when we want to print the value of i. 
  10) At line 12, we will print the length of the prices array that was passed into the function. We print the value of 3. Even though the length variable is of type const, it was declared at the top of the function and not within any specific block. So that valuable will be reachable once we want to print it at line 12. 
  11) 

  
